rfcs/execute_script.md

```diff
@@ -0,0 +1,69 @@
+# RFC 89: `testdriver` Add an `execute_script` method.
+
+## Summary
+
+Add an `execute_script` method to testdriver that takes the source
+text of the script, an async argument, and the context id to execute
+it it. This follows WebDriver semantics, and the returned promise is
+resolved once the script has a result.
+
+## Details
+
+Sometimes a test window wants to check the status of something in a
+remote window. It's possible to do this in a number of ways, for
+example by posting messages from the remote window to the test
+window. However in cross-origin cases messge passing on the js side
+can be difficult or impossible. In addition when tests are defined in
+the test window, it's useful to avoid putting complex logic in other
+browsing contexts because these won't have the same error handling
+properties as the test window.
+
+A simple way to solve these problems is to provide a mechansim to
+execute script in a different context. This isn't possible using
+content APIs alone, but is possible in WebDriver, and we can reuse
+that to define a `execute_script` function that works across contexts:
+
+```
+execute_script(body, async, context)
+```
+
+`body` is a function body for the script to execute. `async` controls
```

Comment:
`async` as an argument seems somewhat painful. Most likely, it'll always be a bare boolean constant at the call sites, which doesn't tend to be readable. Would adding a separate `execute_async_script` method be much more work?

Reply:
Agreed. I'm going to update the RFC with the proposal in https://github.com/web-platform-tests/rfcs/pull/89#issuecomment-888369412